src/components/DataFilter.js

```diff
@@ -0,0 +1,141 @@
+import React, {Component} from 'react';
+import ReactDOM from 'react-dom';
+import {observable} from 'mobx';
+import {observer} from 'mobx-react';
+import {isPlainObject} from 'lodash';
+import cx from 'classnames';
+
+@observer
+export default class DataFilter extends Component {
+  static propTypes = {
+    name: React.PropTypes.string.isRequired,
+    onChange: React.PropTypes.func.isRequired,
+    title: React.PropTypes.string,
+    value: React.PropTypes.string,
+    showOptionsFilter: React.PropTypes.bool
+  }
+
+  static defaultProps = {
+    value: '',
+    showOptionsFilter: true
+  }
+
+  @observable optionsFilter = ''
+  @observable isOpen = false
+
+  componentDidMount() {
+    document.addEventListener('click', this.onDocumentClick);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.onDocumentClick);
+  }
+
+  onDocumentClick = (e) => {
+    if (this.isOpen && !ReactDOM.findDOMNode(this.refs.filter).contains(e.target)) {
+      this.toggle(false);
+    }
+  }
+
+  filterOptions = (e) => {
+    this.optionsFilter = e.target.value;
+  }
+
+  toggle = (isOpen) => {
+    this.isOpen = isOpen;
+  }
+
+  closeOnEscapeKey = (e) => {
+    if (e.key === 'Escape') this.toggle(false);
+  }
+
+  render() {
+    const {
+      name, title, value, options, disabled, onChange, showOptionsFilter
+    } = this.props;
+
+    if (options) {
+      const normalizedOptions = options.map(
+        (option) => isPlainObject(option) ? option : {value: option, title: option}
+      );
+      const selectedOption = normalizedOptions.find((option) => option.value === value);
+      const filteredOptions = this.optionsFilter ?
```

Comment:
filteredOptions could also be a computed property to avoid recalculation on every render